### ISSUE TITLE
docs: characterize rembg mask-level failure modes and lock model success criteria

### ISSUE THREAD

**Issue by shaoster** [shaoster/glaze#413]
## Problem / Motivation

Milestone [#2 — Custom Pottery Crop Model](https://github.com/shaoster/glaze/milestone/2) replaces today's off-the-shelf rembg/u2net pipeline (`tools/piece_image_crop_service.py`, `api/utils.py:calculate_subject_crop`) with a pottery-tuned Ultralytics YOLO-seg model. Before we commit annotation effort, training compute, and a multi-version Modal deployment topology, we need a clear, evidence-backed answer to:

> Where, specifically, does rembg fail on pottery images **at the mask level**, and what bar must the trained replacement clear to be worth shipping?

The trained model's native output is a **foreground segmentation mask** (the bbox is a derivation — see the milestone's locked design decisions). Today's "bad crop" intuition is bbox-shaped: "the crop is too tight" or "the crop cut off the handle." That framing is insufficient for evaluating a mask-producing model. A bbox can look acceptable while the underlying rembg mask is bleeding into the background, dropping a translucent rim, or hallucinating a second subject — and conversely, a slightly off-center bbox can sit on top of a perfectly correct mask. We need the diagnosis and the success criteria to live in **mask space**, the same abstraction layer where the new model operates.

This issue is the analysis ticket that turns the data accumulated by [#421 — Bad-Crop Tagging & Inference Run Persistence](https://github.com/shaoster/glaze/issues/421) (human-submitted `CropRun` rows where `source.type = "human"`, including persisted `mask_asset` references) into (a) a written characterization of rembg's systemic failure classes and (b) the **locked success criteria** that the evaluation harness in [#417 — Evaluation Harness & Baseline Report](https://github.com/shaoster/glaze/issues/417) will judge the trained model against.

The failure taxonomy is **open-ended and data-driven** — there is no fixed category enum in the data model. Human reporters express failure context via the `notes` field on their submitted `CropRun` rows, plus the corrected mask itself. This issue synthesizes those signals into a written taxonomy.

## Proposed Solution

Produce a written report, committed under `docs/agents/` (suggested path: `docs/agents/rembg-failure-modes.md`), that:

1. **Defines the analysis unit as the mask, not the bbox.** Every figure, every metric, and every category in the report compares rembg's predicted mask against a reference (annotator-corrected mask from [#420](https://github.com/shaoster/glaze/issues/420), or human-submitted `CropRun.mask_asset` where available, or the reporter's `notes` where pixel-accurate references are not yet available). Bbox-level observations may be cited as secondary evidence but cannot stand alone.

2. **Characterizes systemic failure classes** drawn from the corpus of human-submitted `CropRun` rows accumulated via [#421](https://github.com/shaoster/glaze/issues/421). Expected categories (refine based on what the data actually shows):
   - Translucent / transmissive glaze (mask under-fills the vessel where light passes through)
   - Low-contrast backgrounds (mask boundary drifts onto background)
   - Multi-subject framing (mask captures secondary objects, props, hands)
   - Holloware vs. flatware geometry (mask fails on rim ellipses, foot rings, plate edges differently than on cups/bowls)
   - Busy or textured backgrounds (mask leaks into background texture)
   - Any additional class the data surfaces (shadows, reflections, dark-on-dark, etc.)

   For each class: representative examples (source image + rembg `mask_asset` + human-corrected reference), a count or share of the flagged corpus, and a short description of the mask-level failure mode (over-segmentation, under-segmentation, boundary drift, hole-fill error, etc.).

3. **Locks success criteria** for [#417](https://github.com/shaoster/glaze/issues/417). These must be expressed in mask space and committed to in this issue's deliverable so [#417](https://github.com/shaoster/glaze/issues/417) does not get to relitigate them. At minimum:
   - A target **mask IoU** threshold on the held-out test set (overall).
   - Per-class IoU floors for the failure categories identified above (so the new model cannot win on aggregate while regressing on translucent glaze, for example).
   - An explicit reduction target for the rate of human-submitted `CropRun` corrections from owners and admins on the flagged corpus.
   - A go/no-go decision rule: which thresholds are mandatory vs. aspirational.

4. **Methodology section** documenting: corpus size, time window, sampling strategy, how human-submitted `CropRun` rows were used as the primary signal, and known biases of the corpus.

## Data Model Reference

The corpus for this analysis is queried from `CropRun` (introduced in [#421](https://github.com/shaoster/glaze/issues/421)):

```python
# Human-submitted corrections (the "flagged bad" signal)
CropRun.objects.filter(source__type="human").select_related("image")

# Automated rembg runs (the baseline to compare against)
CropRun.objects.filter(source__type="automated", source__backend="rembg-u2net")
```

`CropRun.mask_asset` stores `{cloud_name, cloudinary_public_id}` pointing to an RGBA PNG with RGB channels zeroed (alpha = foreground confidence), uploaded to Cloudinary by `run_crop_inference()` in `api/utils.py`. `CropRun.notes` holds free-form reporter context.

## Acceptance Criteria

- [ ] Report committed under `docs/agents/` (e.g. `docs/agents/rembg-failure-modes.md`).
- [ ] Analysis explicitly framed at the mask level; every failure category is described in terms of mask defects, not bbox defects. The report states this framing in its opening and justifies it by reference to the milestone's locked design decision that the new model outputs masks.
- [ ] Failure-class taxonomy is derived from human-submitted `CropRun` rows (and their `notes` + `mask_asset`) persisted by [#421](https://github.com/shaoster/glaze/issues/421), not invented a priori. Each class is supported by representative examples and a count/share of the corpus.
- [ ] Success criteria section is **locked-in**: explicit numerical thresholds (overall mask IoU, per-class IoU floors, bad-crop-report reduction target) plus a go/no-go decision rule. [#417](https://github.com/shaoster/glaze/issues/417) consumes these as fixed inputs.
- [ ] Methodology section documents corpus size, time window, reference-mask source, and known biases.
- [ ] No code changes in this issue's PR. Deliverable is documentation only.

## Out of Scope

- Building the evaluation harness itself (that is [#417](https://github.com/shaoster/glaze/issues/417)).
- Annotating the full training corpus (that is [#420 — Supervised Crop Annotation UI](https://github.com/shaoster/glaze/issues/420) and [#414 — Label Studio Seed-Labeling Bootstrap](https://github.com/shaoster/glaze/issues/414)).
- Any change to `piece_image_crop_service.py`, `api/utils.py`, or `api/tasks.py:detect_subject_crop`.
- Adding a fixed failure-category enum to the data model — the taxonomy lives in this document, not in `CropRun`. If a future issue needs a structured enum, file a follow-up.
- Privacy Policy verification for training-data use (that is part of [#415 — Training Dataset Export Pipeline](https://github.com/shaoster/glaze/issues/415)'s DoD).

## Milestone Cross-References

- Depends on [#421 — Bad-Crop Tagging & Inference Run Persistence](https://github.com/shaoster/glaze/issues/421)
- Blocks [#417 — Evaluation Harness & Baseline Report](https://github.com/shaoster/glaze/issues/417)

Part of milestone [#2 — Custom Pottery Crop Model](https://github.com/shaoster/glaze/milestone/2).
